src/client/Microsoft.Identity.Client/Internal/Requests/Silent/SilentRequest.cs

```diff
@@ -66,20 +66,12 @@ protected override async Task<AuthenticationResult> ExecuteAsync(CancellationTok
                     _logger.Info("Broker is configured and enabled, attempting to use broker instead.");
                     var brokerResult = await _brokerStrategyLazy.Value.ExecuteAsync(cancellationToken).ConfigureAwait(false);
 
+                    // fallback to local cache if broker fails

```

Comment:
if broker is not installed, why is the flag `isBrokerConfigured` set to true?

Reply:
Just my 2 cents. Inferred from [LennoxP90's repro steps](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/4209#issuecomment-1632998180), the flag `isBrokerConfigured` in MSAL .Net is presumably only determined by whether the app developer opted in to use broker, not by whether the broker is installed.

Regardless of the implementation detail above, my higher-level question is, does the fact that we are reviewing implementation details imply that we all agree with the direction of this PR, quoted again from the PR's description? @bgavrilMS 

> Changes proposed in this request
WithBroker = true, interactive token call falls back correctly to browser and completes the flow. **With silent token call for the same account, currently there is no fallback to local cache if broker fails to return result. BrokerStrategy can fail for example if the broker is not available in the machine. The fix is to fall back to local cache for silent token call** as well similar to interactive call.

Reply:
@rayluo - actually `isBrokerConfigured` in this context is `isBrokerEnabled by app developer` AND `is broker available on the OS`. I would have expected the `is broker available on the OS` to fail. 

I agree that the flag is poorly named, but it's just a local variable.

Reply:
Otherwise yes, I agree that silent auth should try to use MSAL's token cache ("client strategy") if broker is not available. 

This is different from the following scenario: 

- app users browser and this puts an RT into the cache
- app then moves to use broker
- silent auth is called

Expected: app MUST ignore the cache as long as the broker is available

The reason for this is that the browser RT is not bound (it's not PRT) and it will break the CA policy related to broker.

Reply:
> I agree that the flag is poorly named, but it's just a local variable.

Should be renamed then. #CleanCode